src/wallet/transaction.h

```diff
@@ -231,6 +241,15 @@ class CWalletTx
         if (nTimeSmart) {
             mapValueCopy["timesmart"] = strprintf("%u", nTimeSmart);
         }
+        if (!m_foreign_outputs.empty()) {
+            std::string s(m_foreign_outputs.size(), 0);
+            for (auto i = m_foreign_outputs.size(); i--; ) {
+                if (m_foreign_outputs.at(i)) {
+                    s[i / 8] |= 1 << (i % 8);
+                }
+            }
+            mapValueCopy["fout"] = s;
+        }
```

Comment:
I am really looking forward to learning what this does. For the life of me I can't figure out what you are doing with the string or why this works.

Reply:
[I got an answer on bitcoin stack overflow.](https://bitcoin.stackexchange.com/questions/115208/help-explaining-some-c-code-i-havent-seen-before)